src/rpcblockchain.cpp

```diff
@@ -1007,36 +1007,44 @@ bool AdvertiseBeacon(std::string &sOutPrivKey, std::string &sOutPubKey, std::str
             std::string sName = GlobalCPUMiningCPID.cpid;
             try
             {
-                // Store the key 
-                sMessage = AddContract(sType,sName,sBase);
+
                 // Backup config with old keys like a normal backup
-                std::string sBeaconBackupOldConfigFilename = GetBackupFilename("gridcoinresearch.conf");
-                boost::filesystem::path sBeaconBackupOldConfigTarget = GetDataDir() / "walletbackups" / sBeaconBackupOldConfigFilename;
-                BackupConfigFile(sBeaconBackupOldConfigTarget.string().c_str());
-                StoreBeaconKeys(GlobalCPUMiningCPID.cpid, sOutPubKey, sOutPrivKey);
+                if(!BackupConfigFile(GetBackupFilename("gridcoinresearch.conf")))
+                {
+                    sError = "Failed to backup old configuration file. Beacon not sent.";
+                    return false;
+                }
+
                 // Backup config with new keys with beacon suffix
-                std::string sBeaconBackupNewConfigFilename = GetBackupFilename("gridcoinresearch.conf", "beacon");
-                boost::filesystem::path sBeaconBackupNewConfigTarget = GetDataDir() / "walletbackups" / sBeaconBackupNewConfigFilename;
-                BackupConfigFile(sBeaconBackupNewConfigTarget.string().c_str());
+                StoreBeaconKeys(GlobalCPUMiningCPID.cpid, sOutPubKey, sOutPrivKey);
+                if(!BackupConfigFile(GetBackupFilename("gridcoinresearch.conf", "beacon")))
+                {
+                    sError = "Failed to backup new configuration file. Beacon not sent. And your config file is probably corrupted now.";
+                    return false;
+                }
+
+                // This prevents repeated beacons
+                nLastBeaconAdvertised = nBestHeight;
```

Comment:
this should be below AddContract?. If AddContract fails then they will have to wait 5 blocks to do another attempt for instance if the wallet is locked.